src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -0,0 +1,174 @@
+
+        double sum = 0.0, norm = pairsCount; // for calculating coverage, stored as sum to avoid frequent casting
+
+        int left = (int) metrics.MEDIAN_INSERT_SIZE;
+        int right = left;
+
+        int i = 0, j = 0;             // represent boundaries of bins that needs update
+        long bins[] = new long[10];   // for storing bin boundaries so that less tests performed during each iteration
+
+        while(i<10){
+            Long sz1 = hist.get(left), sz2 = (left!=right)? hist.get(right) : 0L;
+            if(null!=sz1) { sum += sz1; }
+            if(null!=sz2) { sum += sz2; }
+
+            j = (int) (10.*sum/norm);
+            for(int k=i; k<j; ++k){          // only update necessary ones
+                bins[k] = right - left + 1;
+            }
+            i = j;
+            --left;
+            ++right;
+        }
+
+        metrics.WIDTH_OF_10_PERCENT = (int) bins[0];
+        metrics.WIDTH_OF_20_PERCENT = (int) bins[1];
+        metrics.WIDTH_OF_30_PERCENT = (int) bins[2];
+        metrics.WIDTH_OF_40_PERCENT = (int) bins[3];
+        metrics.WIDTH_OF_50_PERCENT = (int) bins[4];
+        metrics.WIDTH_OF_60_PERCENT = (int) bins[5];
+        metrics.WIDTH_OF_70_PERCENT = (int) bins[6];
+        metrics.WIDTH_OF_80_PERCENT = (int) bins[7];
+        metrics.WIDTH_OF_90_PERCENT = (int) bins[8];
+        metrics.WIDTH_OF_99_PERCENT = (int) bins[9];
+    }
+
+    /**
+     * Trim down width of histogram by tolerance.
+     * Mean and SD are severely impacted by outliers, so they are computed here after trimming.
+     * @param HistogramMADTolerance how many MADs from the median will be tolerated when down-trimming
+     */
+    private void trimHistogramAndCollectMeanAndSD(final double HistogramMADTolerance){
+
+        // range of read length that is going to be tolerated when computing histogram, mean and SD, but will not affect other metrics.
+        final int toleranceRange[] = {(int) metrics.MEDIAN_INSERT_SIZE - (int) (HistogramMADTolerance* metrics.MEDIAN_ABSOLUTE_DEVIATION),
+                                      (int) metrics.MEDIAN_INSERT_SIZE + (int) (HistogramMADTolerance* metrics.MEDIAN_ABSOLUTE_DEVIATION)};
+        final JavaRDD<Integer> trimmedLengthRDD = lengthRDD.filter(length -> length>=toleranceRange[0] && length<=toleranceRange[1]);
+        double lengthArr[] = getRDDArray(trimmedLengthRDD);
+
+        DescriptiveStatistics percent = new DescriptiveStatistics(lengthArr);
+
+        metrics.MEAN_INSERT_SIZE   = percent.getMean();
+        metrics.STANDARD_DEVIATION = percent.getStandardDeviation();
+
+        final Map<Integer, Long> temp = trimmedLengthRDD.countByValue();
+        final TreeMap<Integer, Long> hist = new TreeMap<>(temp);
+        // TODO: fix second argument used in ctor, and potentially change how Histogram is constructed from TreeMap
+        htslibHistogram = new Histogram<>("insert_size", "All_Reads.fr_count");
+        for(final Integer sz: hist.keySet()){
```

Comment:
final `int sz: hist.keySet()` <- better yet, rename `sz` to something better like `size`